Add machine image maintenance strategies to GCM (#8275)
gardener/gardener: docs/usage/shoot_maintenance.md

 - There is a higher version available and the Shoot opted-in for automatic version updates.
 - The currently used version is `expired`.

The target version for machine image upgrades is controlled by the `updateStrategy` field for the machine image in the CloudProfile. Allowed update strategies are `patch`, `minor` and `major`.

Gardener (gardener-controller-manager) populates the `lastMaintenance` field in the Shoot status with the maintenance results.

```text
Last Maintenance:
    Description:     Control Plane: Updated Kubernetes version from "1.26.4" to "1.27.1". Reason: Kubernetes version expired - force update required
    State:           Succeeded
    Triggered Time:  2023-07-28T09:07:27Z
```

Additionally, Gardener creates events with the type `MachineImageVersionMaintenance` or `KubernetesVersionMaintenance` on the Shoot describing the action performed during maintenance, including the reason why an update has been triggered.

> ```text

Reviewer: Same here keep it yaml, but format it properly.
```suggestion
```yaml
```

Author: Those are events on a resource, which are not in a yaml-structure? I verbatim (with minor edits) copied them from a resource, to serve as documentation. So I don't think I should format anything here.

```
LAST SEEN   TYPE      REASON                           OBJECT          MESSAGE
30m         Normal    MachineImageVersionMaintenance   shoot/local     Worker pool "local": Updated image from 'coreos' version 'xy' to version 'abc'. Reason: Automatic update of the machine image version is co
nfigured (image update strategy: major).

30m         Normal    KubernetesVersionMaintenance     shoot/local     Control Plane: Updated Kubernetes version from "1.26.4" to "1.27.1". Reason: Kubernetes version expired - force update required.

15m         Normal    KubernetesVersionMaintenance     shoot/local     Worker pool "local": Updated Kubernetes version '1.26.3' to version '1.27.1'. Reason: Kubernetes version expired - force update required.

```

Otherwise, please make suggestion how to format this to yaml.

 - There is a higher version available and the Shoot opted-in for automatic version updates.
 - The currently used version is `expired`.

> The target version for machine image upgrades depends on the `updateStrategy` (update `{patch, minor, major}`) for the machine image in the CloudProfile.

Reviewer 2: ```suggestion
The target version for machine image upgrades is controlled by the `updateStrategy` field for the machine image in the CloudProfile. Allowed update strategies are `patch`, `minor` and `major`.
```

> Gardener populates the `Last Maintenance` field in the Shoot status with the maintenance results

Reviewer 2: ```suggestion
Gardener (gardener-controller-manager) populates the `lastMaintenance` field in the Shoot status with the maintenance results.
```

```yaml

> Last Maintenance:
>     Description:     Control Plane: Updated Kubernetes version from "1.26.4" to "1.27.1". Reason: Kubernetes version expired - force update required
>     State:           Succeeded
>     Triggered Time:  2023-07-28T09:07:27Z

Reviewer 2: The section is marked as `yaml`:

```suggestion
status:
  lastMaintenance:
    description: "Control Plane: Updated Kubernetes version from "1.26.4" to "1.27.1". Reason: Kubernetes version expired - force update required"
    state: Succeeded
    triggeredTime: "2023-07-28T09:07:27Z"
```

Reviewer: This comment was about formatting the above line in proper yaml.

```

Additionally, Gardener creates events with the type `MachineImageVersionMaintenance` or `KubernetesVersionMaintenance` on the Shoot describing the action performed during maintenance, including the reason why an update has been triggered.

> ```yaml

Reviewer 2: ```suggestion
```text
```
